pkg/apps/metrics/prometheus/metrics.go

```diff
@@ -113,6 +114,12 @@ func (c *appsCollector) Collect(ch chan<- prometheus.Metric) {
 				continue
 			}
 			if util.IsCompleteDeployment(d) {
+				// If a completed rollout is found AFTER we recorded a failed rollout,
+				// do not record the lastFailedRollout as the latest rollout is not
+				// failed.
+				if _, hasFailedRollout := latestFailedRollouts[d.Namespace+"/"+dcName]; hasFailedRollout {
```

Comment:
key := d.Namespace+"/"+dcName

Reply:
yeah, I was thinking about it as well ;-)